core/src/subject/SubjectEntity.ts

```diff
@@ -0,0 +1,250 @@
+import { PerspectiveProxy } from "../perspectives/PerspectiveProxy";
+import { makeRandomPrologAtom } from "./SDNADecorators";
+import { singularToPlural } from "./util";
+
+const flattenPrologList = (list: object): any[] => {
+  let result = []
+  while (list && list["head"]) {
+    result.push(list["head"])
+    list = list["tail"]
+  }
+  return result
+}
+
+export type QueryPartialEntity<T> = {
+  [P in keyof T]?: T[P] | (() => string);
+};
+
+
+export class SubjectEntity {
+  #baseExpression: string;
+  #subjectClass: string;
+  #perspective: PerspectiveProxy
+  author: string;
+  timestamp: string;
+
+  constructor(perspective: PerspectiveProxy, baseExpression?: string) {
+    this.#baseExpression = baseExpression ? baseExpression : `entry://${makeRandomPrologAtom(24)}`;
+    this.#perspective = perspective
+  }
+
+  get baseExpression() {
+    return this.#baseExpression
+  }
+
+  private async getData(id?: string) {
+    const tempId = id ?? this.#baseExpression;
+    let isInstance = await this.#perspective.isSubjectInstance(tempId, this.#subjectClass)
+    if (!isInstance) {
+      throw `Not a valid subject instance of ${this.#subjectClass} for ${tempId}`
+    }
+
+    const subject = await this.#perspective.getSubjectProxy(tempId, this.#subjectClass);
+
+    if (!subject) {
```

Comment:
Don't really understand what this is for.. ?
`subject` isn't used below. if the base is not an instance of the class (which createSubject() would fix) we already jumped out of this function with the throw above...